websites/S/Streamlabs/presence.ts

```diff
@@ -1,65 +1,172 @@
 const presence = new Presence({
 		clientId: "711871296346128395"
 	}),
-	browsingTimestamp = Math.floor(Date.now() / 1000),
-	userType = [
-		"Viewing the ",
-		"Viewing their ",
-		"Modifying their ",
-		"Creating their "
-	];
+	browsingTimestamp = Math.floor(Date.now() / 1000);
 
 let strings,
+	showButtons,
+	path,
+	hash,
 	clipTitle,
 	rewardString,
 	titleSiteCreator,
-	titleDashboard = "Dashboard";
+	titleDashboard = "Dashboard",
+	postTitle;
 
 presence.on("UpdateData", async () => {
 	strings = await presence.getStrings({
 		play: "presence.playback.playing",
 		pause: "presence.playback.paused"
 	});
+	showButtons = await presence.getSetting("showButtons");
+	path = document.location.pathname.split("/");
+	hash = document.location.hash.split("/");
+
 	const presenceData: PresenceData = {
 		largeImageKey: "logo"
 	};
 
-	switch (window.location.hostname) {
-		case "howto.streamlabs.com":
-			presenceData.startTimestamp = browsingTimestamp;
-			presenceData.details = `${userType[0]}Forums`;
-			break;
-		case "support.streamlabs.com":
-			presenceData.startTimestamp = browsingTimestamp;
-			presenceData.details = `${userType[0]}Support Pages`;
-			break;
-		case "dev.streamlabs.com":
-			presenceData.startTimestamp = browsingTimestamp;
-			presenceData.details = `${userType[0]}API Documentation`;
-			break;
-		case "ideas.streamlabs.com":
-			presenceData.startTimestamp = browsingTimestamp;
-			presenceData.details = `${userType[0]}Feature Suggestions`;
-			break;
-	}
-
-	if (window.location.hostname === "streamlabs.com") {
-		switch (document.location.pathname) {
-			case "/":
+	if (window.location.hostname === "dev.streamlabs.com") {
+		presenceData.startTimestamp = browsingTimestamp;
+		presenceData.details = "Viewing the API Documentation";
+	} else if (window.location.hostname === "streamlabs.com") {
+		switch (path[1]) {
+			case "":
 				presenceData.startTimestamp = browsingTimestamp;
 				presenceData.details = "Home Page";
 				break;
-			case "/login":
+
+			case "login":
 				presenceData.startTimestamp = browsingTimestamp;
 				presenceData.details = "Logging in";
 				break;
-			case "/clips":
+
+			case "clips":
+				presenceData.startTimestamp = browsingTimestamp;
+				presenceData.details = "Viewing their Clips";
+				break;
+
+			case "best-donation-clips":
+				presenceData.startTimestamp = browsingTimestamp;
+				presenceData.details = "Viewing the Best Clips";
+				break;
+
+			case "content-hub":
 				presenceData.startTimestamp = browsingTimestamp;
-				presenceData.details = `${userType[1]}Clips`;
+				presenceData.details = "Viewing the Content Hub";
+
+				if (path[2] === "post") {
+					postTitle = document
+						.querySelector(".content-card__title")
+						.firstChild.textContent.trim();
+					presenceData.state = `${postTitle}`;
```

Comment:
this is already a string and it doesn't seem to be used after this assignment, so you can just assign it directly
```suggestion
					presenceData.state = document
						.querySelector(".content-card__title")
						.firstChild.textContent.trim();
```